plugins/discovery-azure/src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeServiceImpl.java

```diff
@@ -97,16 +89,62 @@ public HostedServiceGetDetailedResponse getServiceDetails() {
+    protected void doStart() {
+        logger.trace("creating new Azure client for [{}], [{}]", subscriptionId, serviceName);
+
+        try {
+            Configuration configuration = ManagementConfiguration.configure(new URI(Azure.ENDPOINT),
+                subscriptionId, keystorePath, keystorePassword, keystoreType);
+            computeManagementClient = ComputeManagementService.create(configuration);
+        } catch (IOException|URISyntaxException e) {
+            logger.error("can not start azure client: {}", e.getMessage());
+            throw new ElasticsearchException("can not start azure client", e);
+        }
+
+        // That's a shame. Apparently ComputeManagementService catches and traces a FileNotFoundException
+        // but swallows it.
+        // So for now we send a request which will throw an error and stop elasticsearch
+        // TODO Remove this if https://github.com/Azure/azure-sdk-for-java/issues/565 is fixed
+        try {
+            logger.debug("checking azure client...");
+            HostedServiceListResponse list = computeManagementClient.getHostedServicesOperations().list();
+            if (list.getHostedServices().isEmpty()) {
+                // We should have at least one service.
+                throw new ElasticsearchException("no hosted service seems to be available for your azure account.");
+            }
+            logger.debug("Known hosted services for your azure account:");
+            boolean serviceFound = false;
+            for (HostedServiceListResponse.HostedService hostedService : list) {
+                logger.debug("- {}", hostedService.getServiceName());
+                if (hostedService.getServiceName().equals(serviceName)) {
+                    serviceFound = true;
+                }
+            }
+
+            if (serviceFound == false) {
+                throw new ElasticsearchException("[{}] does not exist for your azure account. " +
+                    "To list available services, run in debug mode or check your azure console at https://portal.azure.com/.",
+                    serviceName);
+            }
+        } catch (ServiceException|ParserConfigurationException|URISyntaxException|SAXException|IOException e) {
+            // Logging here the message only. The full stacktrace is then printed because
+            // AzureDiscoveryModule won't start and guice will print the full stack.
+            logger.error("can not start azure client. Check your elasticsearch configuration. {}", e.getMessage());
+            throw new ElasticsearchException("can not start azure client. Check your elasticsearch configuration. {}", e, e.getMessage());
+        }
     }
 
+    // This method is never called implicitly!
     @Override
     protected void doStop() throws ElasticsearchException {
     }
 
+    // This method is never called implicitly!
     @Override
     protected void doClose() throws ElasticsearchException {
+        // This code is never executed!
```

Comment:
Well. I guess I added there 2 years ago before knowing that it won't be ever executed! 
I did not remove it as I think we should may be find a better solution when the node stops to actually close the azure client.
I think this is not related to this PR though. But I should probably add a TODO in my comment.